pkg/webhook/tls_wrapper.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"crypto/tls"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/go-logr/logr"
+)
+
+type wrappedTLS struct {
+	sync.Mutex
+	certificate *tls.Certificate
+	log         logr.Logger
+}
+
+func (w *wrappedTLS) setLogger(logger logr.Logger) error {
+	w.log = logger
+	return nil
+}
+
+func (w *wrappedTLS) getCertificate(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
+	w.Lock()
+	defer w.Unlock()
+
+	return w.certificate, nil
+}
+
+func (w *wrappedTLS) loadCertificate(cert, key string) error {
+	w.Lock()
+	defer w.Unlock()
+
+	certKey, err := tls.LoadX509KeyPair(cert, key)
+	if err != nil {
+		return err
+	}
+
+	w.certificate = &certKey
+	return nil
+}
+
+func (w *wrappedTLS) autoloadTLS(cert, key string) error {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return err
+	}
+	defer watcher.Close()
+
+	done := make(chan error)
+	go func() {
+		for {
+			select {
+			case event := <-watcher.Events:
+				if event.Op&fsnotify.Write == fsnotify.Write {
```

Comment:
I doubt if it will work, since when a secret get updated it actually triggers a rename event. Ref: [algorithm of volume writer](https://github.com/kubernetes/kubernetes/blob/b5c34d0c436dc6b95f9b78769d090af1bbbfa629/pkg/volume/util/atomic_writer.go#L89-L119).